packages/jest-runtime/src/index.ts

```diff
@@ -563,10 +581,78 @@ class Runtime {
     }
   }
 
-  getAllCoverageInfoCopy() {
+  async collectV8Coverage() {
+    this._v8CoverageInstrumenter = new CoverageInstrumenter();
+
+    await this._v8CoverageInstrumenter.startInstrumenting();
+  }
+
+  async stopCollectingV8Coverage() {
+    if (!this._v8CoverageInstrumenter) {
+      throw new Error('You need to call `collectV8Coverage` first.');
+    }
+    this._v8CoverageResult = await this._v8CoverageInstrumenter.stopInstrumenting();
+  }
+
+  async getAllCoverageInfoCopy() {
+    if (this._v8CoverageResult) {
+      return this._mapV8CoverageToIstanbul();
+    }
     return deepCyclicCopy(this._environment.global.__coverage__);
   }
 
+  private async _mapV8CoverageToIstanbul() {
+    if (!this._v8CoverageResult) {
+      throw new Error('You need to `stopCollectingV8Coverage` first');
+    }
+    const filtered = this._v8CoverageResult
+      .filter(res => res.url.startsWith('file://'))
+      .map(res => ({...res, url: res.url.replace('file://', '')}))
+      .filter(res => this._fileTransforms.get(res.url))
+      .filter(res =>
+        shouldInstrument(res.url, this._coverageOptions, this._config),
+      );
+
+    const result = await Promise.all(
+      filtered.map(async res => {
+        // this is safe since we filter out those mossing this above
```

Comment:
There you go. Sorry I missed this comment.

```suggestion
        // this is safe since we filter out those missing this above
```